### PR TITLE
Standardize network settings page

### PR DESCRIPTION
**Pull request by rekmarks**
### Summary
Our network settings page has been janky for a long time. This PR attempts to update its UI such that it's inline with our other settings pages. This was accomplished by the creation of an additional route used only in the popup, for the network form.

My sense is that the CSS of this page is still probably less than great, but it ought to do for now. Reviewers should pay extra attention to my use of routes and `history`, since I'm inexperienced with `react-router-dom`.

These changes mostly affect the popup view; the fullscreen view should be largely unaffected.

### Changes in Detail
- The browser back functionality now works correctly with the network form
  - I also added the settings page header back arrow to the networks settings, and it works
- The network settings subheader was removed in the popup
- The "Add Network" button was added to a sticky footer in the networks settings page
- Network form buttons are now hidden when the form is `viewOnly`
- Network settings routes were updated
  - There's a new route for the network form, which is only used in the popup
  - The fullscreen UI always renders the form, and the new route is never used there

### Testing Steps
  - Open the network form, in fullscreen and the popup
  - Try to break it

### Screenshots / Recordings
<details>
<summary>Before</summary>

</details>

<details>
<summary>After</summary>

</details>

<details>
<summary>After (Recording)</summary>
Note: The weird shadows that stick around are just an artifact of the recording.

</details>

### CODE REVIEW

File: ui/app/pages/settings/networks-tab/networks-tab.component.js [hunk 1]
```diff
@@ -208,49 +200,61 @@ export default class NetworksTab extends PureComponent {
...
                 chainId={chainId}
                 ticker={ticker}
-                onClear={() => {
+                onClear={(shouldUpdateHistory = true) => {
                   setNetworksTabAddMode(false)
-                  setSelectedSettingsRpcUrl(null)
+                  setSelectedSettingsRpcUrl('')
+                  if (shouldUpdateHistory && !isFullScreen) {
+                    history.push(NETWORKS_ROUTE)
+                  }
```
Comment (rekmarks) [inline ui/app/pages/settings/networks-tab/networks-tab.component.js:208]: This spares us from re-pushing `NETWORKS_ROUTE` unnecessarily. Used in `NetworkForm.componentWillUnmount`.

File: ui/app/pages/settings/networks-tab/networks-tab.component.js [hunk 2]
```diff
@@ -30,10 +30,13 @@ export default class NetworksTab extends PureComponent {
...
     networkDefaultedToProvider: PropTypes.bool,
+    history: PropTypes.object.isRequired,
+    shouldRenderNetworkForm: PropTypes.bool.isRequired,
+    isFullScreen: PropTypes.bool.isRequired,
   }
 
-  UNSAFE_componentWillMount () {
-    this.props.setSelectedSettingsRpcUrl(null)
+  componentWillUnmount () {
+    this.props.setSelectedSettingsRpcUrl('')
```
Comment (rekmarks) [inline ui/app/pages/settings/networks-tab/networks-tab.component.js:39]: The default state is `''`, and resetting it on unmount works equally well.

**Review by brad-decker — APPROVED** [review]
LGTM!

**Comment by brad-decker** [comment]
Works like a charm on my machine. 👍

**Review by rachelcope — APPROVED** [review]
This looks great. Great improvements to make it consistent with the rest of the settings pages

**Pull request merged**